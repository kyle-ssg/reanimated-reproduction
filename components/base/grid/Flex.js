// propTypes: value: OptionalNumber
import React, { PureComponent } from "react";
import propTypes from "prop-types";

const cn = require("classnames");

const _propTypes = {
  /** The element's children */
  children: propTypes.node,
  /** The element's class name */
  className: propTypes.string,
  /** The flex value */
  value: propTypes.number,
};

/**
 * Div with flex
 */
export class Flex extends PureComponent {
  static displayName = "Flex";

  static propTypes = _propTypes;

  static defaultProps = {
    value: 1,
  };

  render() {
    const {
      props: { className, children, value, ...rest },
    } = this;
    return (
      <div
        {...rest}
        className={cn(
          {
            flex: true,
          },
          `flex-${value}`,
          className
        )}
      >
        {children}
      </div>
    );
  }
}

global.Flex = Flex;
export default Flex;
