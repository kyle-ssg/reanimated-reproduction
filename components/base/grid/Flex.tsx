import React from 'react'
import cn from 'classnames'

export type FlexType = {
  className?: string
  value?: number
  onClick?: () => void
}

//Div with flex
const Flex: React.FC<FlexType> = ({ className, value = 1, ...props }) => (
  <div {...props} className={cn({ flex: true }, `flex-${value}`, className)} />
)

Flex.displayName = 'Flex'
global.Flex = Flex
export default Flex
