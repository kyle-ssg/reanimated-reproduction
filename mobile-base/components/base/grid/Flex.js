//propTypes: value: OptionalNumber

module.exports =
    class extends React.Component {
        render() {
            return (
                <View style={[this.props.style, {flex:this.props.value || 1}]}>
                    {this.props.children}
                </View>
            )
        }
    };