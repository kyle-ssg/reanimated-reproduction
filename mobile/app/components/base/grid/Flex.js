// propTypes: value: OptionalNumber

const Flex = class extends React.Component {
    render() {
        return (
            <Animated.View
              style={[this.props.style, { flex: this.props.value }]}
              testID={this.props.testID}
            >
                {this.props.children}
            </Animated.View>
        );
    }
};

Flex.defaultProps = {
    value: 1,
};

Flex.propTypes = {
    value: OptionalNumber,
    children: OptionalElement,
    style: React.PropTypes.any,
    testID: OptionalString,
};

module.exports = Flex;
