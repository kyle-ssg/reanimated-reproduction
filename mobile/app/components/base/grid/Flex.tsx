import React from 'react'
import { View } from 'react-native'

interface Props {
  accessible?: boolean
  accessibilityLabel?: string
  value?: number
  children?: React.ReactNode
  style?: ReactNative.ViewStyle
  testID?: string
  space?: boolean
}

const Flex: React.FC<Props> = ({
  value = 1,
  accessible,
  accessibilityLabel,
  style,
  space,
  testID,
  children,
}) => (
  <View
    accessible={accessible}
    accessibilityLabel={accessibilityLabel}
    style={[
      style,
      { flex: value },
      space && { justifyContent: 'space-between' },
    ]}
    testID={testID}
  >
    {children}
  </View>
)

export default Flex
