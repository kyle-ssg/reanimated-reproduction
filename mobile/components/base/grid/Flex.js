//propTypes: value: OptionalNumber

const Flex = class extends React.Component {
  render () {
    return (
      <View style={[this.props.style, { flex: this.props.value }]}>
        {this.props.children}
      </View>
    );
  }
};

Flex.defaultProps = {
  value: 1
};

Flex.propTypes = {
  value: OptionalNumber,
  children: OptionalElement,
  style: React.PropTypes.any
};

module.exports = Flex;
