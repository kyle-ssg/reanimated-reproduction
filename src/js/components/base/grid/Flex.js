//propTypes: value: OptionalNumber

const Flex = class extends React.Component {
  render () {
    return (
      <div {... this.props} className={(this.props.className||'') + ' flex flex-1'}>
        {this.props.children}
      </div>
    );
  }
};

Flex.defaultProps = {
  value: 1
};

Flex.propTypes = {
  value: OptionalNumber,
  children: OptionalElement,
  style: React.PropTypes.any
};

module.exports = Flex;
