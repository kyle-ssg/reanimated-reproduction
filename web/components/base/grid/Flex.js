// propTypes: value: OptionalNumber
import React from 'react';
import propTypes from 'prop-types';

const cn = require('classnames');

const Flex = ({ className, children, value, ...rest }) => (
    <div
      {...rest}
      className={cn({
          'flex': true,
      }, `flex-${value}`, className)}
    >
        {children}
    </div>
);

Flex.displayName = 'Flex';

Flex.defaultProps = {
    value: 1,
};

Flex.propTypes = {
    className: propTypes.string,
    value: propTypes.number,
    children: propTypes.node,
    style: propTypes.any,
};

window.Flex = Flex;