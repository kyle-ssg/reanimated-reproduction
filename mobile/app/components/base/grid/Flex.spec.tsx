import * as React from 'react'
import Flex from './Flex'
import renderer from 'react-test-renderer'

describe('<Flex />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<Flex />)
    expect(tree).toMatchSnapshot()
  })
})
