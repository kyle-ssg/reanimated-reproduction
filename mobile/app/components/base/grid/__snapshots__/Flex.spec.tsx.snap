// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
